controller/setting_controller.go

```diff
@@ -390,6 +395,85 @@ func (sc *SettingController) syncBackupTarget() (err error) {
 	return nil
 }
 
+func (sc *SettingController) handleSecretsForAWSIAMRoleAnnotation(backupTargetURL, oldSecretName, newSecretName string, isBackupTargetURLChanged bool) (err error) {
+	isSameSecretName := oldSecretName == newSecretName
+	if isSameSecretName && !isBackupTargetURLChanged {
+		return nil
+	}
+
+	isHavingAWSIAMRoleArn := false
+	if !isSameSecretName {
+		isHavingAWSIAMRoleArn, _, err = sc.updateSecretForAWSIAMRoleAnnotation(backupTargetURL, oldSecretName, true)
+		if err != nil {
+			return err
+		}
+	}
+	_, isNewSecretNotExist, err := sc.updateSecretForAWSIAMRoleAnnotation(backupTargetURL, newSecretName, false)
```

Comment:
nit: consider using `isNewSecret` instead to improve readability.

Reply:
Modified to `isValidSecret`. Thanks